http/src/main/java/io/micronaut/http/MediaType.java

```diff
@@ -587,8 +660,14 @@ private String toString0() {
         if (parameters.isEmpty()) {
             return name;
         } else {
-            return name + ";" + parameters.entrySet().stream().map(Object::toString)
-                    .collect(Collectors.joining(";"));
+            StringBuilder sb = new StringBuilder(name);
+            for (Map.Entry<CharSequence, String> e : parameters.entrySet()) {
```

Comment:
Would `parameters.forEach((key, value))` not be more efficient? As far as I understand that removes the need to create the  `Entry` instances

Reply:
Yeah good catch